packages/server/src/models/flower.entity.ts

```diff
@@ -0,0 +1,30 @@
+import { Column, Entity, PrimaryGeneratedColumn, ManyToOne } from 'typeorm';
+import { Actions } from '../constants/actions';
+import Shelf from './shelf.entity';
+
+@Entity()
+export default class Flower {
+  @PrimaryGeneratedColumn()
+  id: number;
+
+  @Column()
+  shelfId: number;
+
+  @Column()
+  order: number;
+
+  @Column()
+  name: string;
+
+  @Column()
+  description: string;
+
+  @Column('simple-array')
+  pictureUrls: string[];
+
+  @Column('simple-json')
+  rrules: { [key in Actions]: string }
+
+  @ManyToOne(() => Shelf, shelf => shelf.flowers)
+  shelf: Shelf;
```

Comment:
We have shelfId and shelf do we need both?

Reply:
Fixed